src/openassetio-core/include/openassetio/errors/exceptions.hpp

```diff
@@ -4,16 +4,36 @@
 
 #include <stdexcept>
 #include <string>
+#include <utility>
 
 #include <openassetio/export.h>
+#include <openassetio/errors/BatchElementError.hpp>
 #include <openassetio/typedefs.hpp>
 
 namespace openassetio {
 inline namespace OPENASSETIO_CORE_ABI_VERSION {
+/**
+ * This namespace contains types related to error handling.
```

Comment:
[consider] Is this now "Exception types" vs general types related to error handling?